File: src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/Semantic/SemanticTokens.ts
/* --------------------------------------------------------------------------------------------
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 * ------------------------------------------------------------------------------------------ */

export class SemanticTokens {
    public readonly resultId?: string;

    public readonly data: Uint32Array;

    constructor(data: Uint32Array, resultId?: string) {
        this.resultId = resultId;
        this.data = data;
    }
}

File: src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/Models/SemanticTokens.cs
// Copyright (c) .NET Foundation. All rights reserved.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

using System.Runtime.Serialization;

namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp.Models
{
    /// <summary>
    /// Class representing response to semantic tokens messages.
    /// </summary>
    [DataContract]
    public class SemanticTokens
    {
        /// <summary>
        /// Gets or sets a property that identifies this version of the document's semantic tokens.
        /// </summary>
        [DataMember(Name = "resultId")]
        public string ResultId { get; set; }

        /// <summary>
        /// Gets or sets and array containing encoded semantic tokens data.
        /// </summary>
        [DataMember(Name = "data", IsRequired = true)]
        public int[] Data { get; set; }
    }
}
